core/src/main/scala/cromwell/core/actor/ThrottlerActor.scala

```diff
@@ -0,0 +1,25 @@
+package cromwell.core.actor
+
+import cats.data.NonEmptyVector
+
+import cats.syntax.traverse._
+import cats.instances.vector._
+import cats.instances.future._
+import scala.concurrent.Future
+import scala.concurrent.duration.Duration
+
+/**
+  * By removing the periodic flush and setting the batch size to 1,
+  * This transforms the BatchActor into throttler processing commands one at a time.
+  */
+abstract class ThrottlerActor[C] extends BatchActor[C](Duration.Zero, 1) {
+  override def weightFunction(command: C) = 1
+  override final def process(data: NonEmptyVector[C]): Future[Int] = {
+    // This ShouldNotBePossible™ but in case it happens, instead of dropping elements process them all anyway
```

Comment:
I'm assuming you're right here, but can you explain why finding yourself in an impossible state means that you should go ahead and proceed normally? That seems like a bad thing?

Reply:
With a batch size of 1 and each element having a weight of 1, the assumption is that only single element vectors will be passed to this function. That is because of the logic implemented in the parent `BatchActor` which calls the `process` function with a maxWeight of batchSize.
That is good because the goal of this class is to make the actor process one element at a time.
However if for some reason the `process` function is being called with more than one element (which would mean the parent `BatchActor` has a bug - which obviously is possible but hopefully not the case),
we don't want to just process the first one and drop the rest as they might be important operations (DB queries most likely). So instead it logs that something's wrong and process them all anyway.